### PR TITLE
Refactor and remove unused code from trace store

### PR DESCRIPTION
**Pull request by nadundesilva**
## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/observability/tracer/TracersStore.java [hunk 1]
```diff
@@ -51,31 +52,29 @@ private TracersStore() {
...
+            String tracerName = configRegistry.getConfigOrDefault(TRACER_NAME_CONFIG, overallProviderName);
 
             ServiceLoader<OpenTracer> openTracers = ServiceLoader.load(OpenTracer.class);
-            HashMap<String, OpenTracer> tracerMap = new HashMap<>();
-            openTracers.forEach(t -> tracerMap.put(t.getName().toLowerCase(), t));
-
-            String defaultReporterName = configRegistry.getConfigOrDefault(CONFIG_OBSERVABILITY_PROVIDER, JAEGER);
-            String tracerName = configRegistry.getConfigOrDefault(TRACER_NAME_CONFIG, defaultReporterName);
+            for (OpenTracer openTracer : openTracers) {
+                if (Objects.equals(openTracer.getName().toLowerCase(), tracerName.toLowerCase())) {
```
Comment (riyafa) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/observability/tracer/TracersStore.java:60]: can't we use `tracerName.equalsIgnoreCase(openTracer.getName())`

Comment (nadundesilva) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/observability/tracer/TracersStore.java]: Fixed

File: misc/tracing-extensions/modules/ballerina-choreo-extension/src/main/java/org/ballerinalang/observe/trace/extension/choreo/ChoreoJaegerReporter.java [hunk 2]
```diff
@@ -126,13 +126,15 @@ private synchronized void append(JaegerSpan jaegerSpan) {
...
+            synchronized (traceSpans) {
+                traceSpans.add(traceSpan);
+            }
         }
 
         @Override
         public void run() {
             ChoreoTraceSpan[] spansToBeSent;
-            synchronized (this) {
+            synchronized (traceSpans) {
```
Comment (Kishanthan) [inline misc/tracing-extensions/modules/ballerina-choreo-extension/src/main/java/org/ballerinalang/observe/trace/extension/choreo/ChoreoJaegerReporter.java:137]: what is the reason for this change? synchronizing on a different object.

Comment (nadundesilva) [reply misc/tracing-extensions/modules/ballerina-choreo-extension/src/main/java/org/ballerinalang/observe/trace/extension/choreo/ChoreoJaegerReporter.java]: This change is not required. I reverted the change in 0b5ba74c9a423930d32f41451d597e6fc1dc2fe4

**Comment by Kishanthan** [comment]
LGTM

**Review by Kishanthan — APPROVED** [review]
